modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItNotNullConstraintTest.java

```diff
@@ -0,0 +1,280 @@
+                view.put(null, Tuple.create(Map.of("id", 1)), Tuple.create());
+            });
+            assertThat(err.getMessage(), containsString("Column 'VAL' does not allow NULLs"));
+        }
+
+        {
+            KeyValueView<Integer, Val> view = table.keyValueView(Integer.class, Val.class);
+            MarshallerException err = assertThrows(MarshallerException.class, () -> {
+                Val val = new Val();
+                view.put(null, 1, val);
+            });
+            assertThat(err.getMessage(), containsString("Column 'VAL' does not allow NULLs"));
+        }
+    }
+
+    @Test
+    @Disabled("https://issues.apache.org/jira/browse/IGNITE-22040")
+    public void testRecordView() {
+        sql("CREATE TABLE kv (id INTEGER PRIMARY KEY, val INTEGER NOT NULL)");
+
+        Table table = CLUSTER.aliveNode().tables().table("KV");
+
+        {
+            RecordView<Rec> view = table.recordView(Rec.class);
+            MarshallerException err = assertThrows(MarshallerException.class, () -> {
+                Rec rec = new Rec();
+                view.insert(null, rec);
+            });
+            assertThat(err.getMessage(), containsString("Column 'ID' does not allow NULLs"));
+        }
+
+        {
+            RecordView<Rec> view = table.recordView(Rec.class);
+            MarshallerException err = assertThrows(MarshallerException.class, () -> {
+                Rec rec = new Rec();
+                rec.id = 42;
+                view.insert(null, rec);
+            });
+            assertThat(err.getMessage(), containsString("Column 'VAL' does not allow NULLs"));
+        }
+
+        {
+            RecordView<Tuple> view = table.recordView();
+            MarshallerException err = assertThrows(MarshallerException.class, () -> {
+                view.insert(null, Tuple.create(Map.of("id", 1)));
+            });
+            assertThat(err.getMessage(), containsString("Column 'VAL' does not allow NULLs"));
+        }
+    }
+
+    @Test
+    @Disabled("https://issues.apache.org/jira/browse/IGNITE-22040")
+    public void testKeyValueViewDataStreamer() {
+        sql("CREATE TABLE kv (id INTEGER PRIMARY KEY, val INTEGER NOT NULL)");
+
+        Table table = CLUSTER.aliveNode().tables().table("KV");
+
+        {
+            KeyValueView<Tuple, Tuple> view = table.keyValueView();
+
+            checkDataStreamer(view, new SimpleEntry<>(Tuple.create(Map.of("id", 1)), Tuple.create()), "VAL");
+        }
+
+
+        {
+            KeyValueView<Integer, Integer> view = table.keyValueView(Integer.class, Integer.class);
+            checkDataStreamer(view, new SimpleEntry<>(1, null), "VAL");
+        }
+
+        {
+            KeyValueView<Integer, Val> view = table.keyValueView(Integer.class, Val.class);
+            Val val = new Val();
+            checkDataStreamer(view, new SimpleEntry<>(1, val), "VAL");
+        }
+    }
+
+    @Test
+    @Disabled("https://issues.apache.org/jira/browse/IGNITE-22040")
```

Comment:
I'm not sure that is a good idea to disable these tests. We can check the current behavior and rework it when IGNITE-22040 is taken.